docs/en/sdk-manual/react-native.md

```diff
@@ -0,0 +1,376 @@
+        _traffic_source_name: "Summer promotion",
+      }
+   });
+   ```
+
+2. Add global attributes after initializing the SDK
+   ``` typescript
+   ClickstreamAnalytics.setGlobalAttributes({
+     _traffic_source_medium: "Search engine",
+     level: 10,
+   });
+   ```
+
+It is recommended to set global attributes when initializing the SDK, global attributes will be included in all events
+that occur after it is set.
+
+#### Delete global attribute
+
+``` typescript
+ClickstreamAnalytics.deleteGlobalAttributes(['level','_traffic_source_medium']);
+```
+
+#### Login and logout
+
+```typescript
+import { ClickstreamAnalytics } from '@aws/clickstream-react-native';
+
+// when user login success.
+ClickstreamAnalytics.setUserId("userId");
+
+// when user logout
+ClickstreamAnalytics.setUserId(null);
+```
+
+#### Add user attribute
+
+```typescript
+import { ClickstreamAnalytics } from '@aws/clickstream-react-native';
+
+ClickstreamAnalytics.setUserAttributes({
+  userName: "carl",
+  userAge: 22
+});
+```
+
+Current login user's attributes will be cached in disk, so the next time app open you don't need to set all user
+attributes again, of course you can use the same API `ClickstreamAnalytics.setUserAttributes()` to update the current
+user's attribute when it changes.
+
+!!! info "Important"
+
+    If your application is already published and most users have already logged in, please manually set the user attributes once when integrate the Clickstream SDK for the first time to ensure that subsequent events contains user attributes.
+
+#### Record event with items
+
+You can add the following code to log an event with an item.
+
+```typescript
+import { ClickstreamAnalytics, Item } from '@aws/clickstream-react-native';
+
+const itemBook: Item = {
+  id: '123',
+  name: 'Nature',
+  category: 'book',
+  price: 99,
+  book_publisher: "Nature Research",
+};
+ClickstreamAnalytics.record({
+  name: 'view_item',
+  attributes: {
+    currency: 'USD',
+    event_category: 'recommended',
+  },
+  items: [itemBook],
+});
+```
+
+For logging more attribute in an item, please refer to [item attributes](android.md#item-attributes).
+
+!!! warning "Important"
+
+    Only pipelines from version 1.1+ can handle items with custom attribute.
+    
+    item id is required attribute, if not set the item will be discarded.
+
+#### Record Screen View events manually
+
+By default, SDK will automatically track the preset `_screen_view` event when Android Activity triggers `onResume` or
+iOS ViewController triggers `viewDidAppear`.
+
+You can also manually record screen view events whether automatic screen view tracking is enabled, add the following
+code to record a screen view event with two attributes.
+
+* `SCREEN_NAME` Required. Your screen's name.
+* `SCREEN_UNIQUE_ID` Optional. Set the id of your component. If you do not set, the SDK will set a default value based
+  on the hashcode of the current native Activity or native ViewController.
```

Comment:
so SCREEN ID is not required for manual tracking screen view? but should we allow user user to set it optionally if they want to

Reply:
The meaning of screen id here is the class path of the screen. The user can set it through add additional custom attributes if needed. For App screen view, I think these two common attributes are enough.